### PR TITLE
Reduce default sample concurrency to two

### PR DESCRIPTION
**Pull request by peppy**
8 was way too high for most use-cases. Also makes this configurable at a SampleManager level if this default is not appropriate for certain use-cases. Can be expanded to per-Get when necessary.

### CODE REVIEW

**Comment by smoogipoo** [comment]
Why was this too high for "most use cases"? Were you experiencing a performance problem or something? What issue did this cause?

Limiting to 2 samples by default for framework seems very limiting. Would rather this was int.MaxValue and osu! lowered it to 2, if overlapping samples were the problem.

**Comment by peppy** [comment]
Yeah, not sure what the best default is. For a sample that is being triggered by a user (ie .for a hover effect), anything more than two rapes the ears, therefore I went with a safe low default rather than leave it up to the framework consumer to adjust every time this comes up.

I can't think of any use-case using higher than two which wouldn't want to be specifying its own limit locally (at the time of usage).

**Comment by smoogipoo** [comment]
Maybe in the menu there's no use-case for higher than two, but gameplay certainly is. Just simply having an object with a whistle + finish addition will give you 3 samples. Add onto that any other hitobject samples, and combo burst sounds... Or mania...

Perhaps better would be to classify UI samples vs non-UI samples, and limit those ones, at an osu! level.

**Comment by peppy** [comment]
this limit is per sample, not global to sample playback.

**Comment by smoogipoo** [comment]
The other way around, this is global. Per-sample would be being required to change the playback concurrency if you desired this limitation.

Globally, I want there to be as many concurrent samples played as allowed by BASS, such that there is no case where we're trying to calculate an upper bound for concurrent playback in gameplay. I would mention other games relying on the framework in the future too, where this sounds like a hidden limitation.

**Comment by peppy** [comment]
As I explained IRL, what I mean is that the concurrency limit is per *sample*. ie. if you have two samples that you loaded individually, each will have a concurrency limit as specified.

File: osu.Framework/Audio/Sample/Sample.cs [hunk 1]
```diff
@@ -5,5 +5,17 @@ namespace osu.Framework.Audio.Sample
 {
     public abstract class Sample : AudioComponent
     {
+        public const int DEFAULT_CONCURRENCY = 2;
+
+        protected int PlaybackConcurrency;
```
Comment (smoogipoo) [inline osu.Framework/Audio/Sample/Sample.cs]: Can this be readonly? There should be no case where PlaybackConcurrency is changed.

Comment (peppy) [reply osu.Framework/Audio/Sample/Sample.cs]: definitely

**Pull request merged**